### PR TITLE
OJ-30026-use-jf-ingest-logging

### PR DESCRIPTION
**Pull request by gavinpitt-jf**
Work to cut over the new standardized logging library to use in Agent.

Need to wait for https://github.com/Jellyfish-AI/jf_ingest/pull/9 to be merged and published on pypi

## Test Notes

1. Build the agent locally, like so: `docker build . -t agent_local`
2. Run the new local agent with the following commands:
```
 DOCKER_HOST="unix://$HOME/.colima/agent/docker.sock"
    mkdir $PWD/jf_agent_output
    docker run --rm \
    --mount type=bind,source=$PWD/config.yml,target=/home/jf_agent/config.yml \
    --mount type=bind,source=$PWD/jf_agent_output,target=/home/jf_agent/output \
    --env-file $PWD/creds.env \
    agent_local --jellyfish-api-base http://host.docker.internal:8000
```
3. Verify output to stdout and to logs
```
mkdir: /Users/gavinpitt/Code/jf_agent/jf_agent_output: File exists
Will write output files into ./output/20231018_131022
Attempting to generate Jira Manifest...
Processing 3 Jira Projects...
Done processing 3 Jira Projects!
Done processing Jira Manifest!
Successfully created Jira Manifest
Attempting to generate Git Manifests...
Generating manifest for instance orthogonal-networks-GH
Processing git instance orthogonal-networks-GH for company orthogonal-networks under github org Jellyfish-AI
Processing 30 repos
Done processing Repos
Processing 103 users
Done processing Users
Attempting upload of 2 manifest(s) to Jellyfish...
Attempting to upload JiraDataManifest_orthogonal-networks_REMOTE_20231018T13:10:27 manifest to s3...
Successfully uploaded JiraDataManifest_orthogonal-networks_REMOTE_20231018T13:10:27 manifest to s3!
Attempting to upload GitDataManifest_orthogonal-networks_orthogonal-networks-GH_Jellyfish-AI_REMOTE_20231018T13:10:30 manifest to s3...
Successfully uploaded GitDataManifest_orthogonal-networks_orthogonal-networks-GH_Jellyfish-AI_REMOTE_20231018T13:10:30 manifest to s3!
Successfully uploaded 2 manifest(s) to Jellyfish!
Obtained Jira configuration, attempting download...
downloading jira fields... ✓
downloading jira projects... ✓
downloading jira project components... ✓
downloading jira versions... ✓
downloading jira users...  ✓
downloading jira resolutions... downloading jira issue types...  ✓
downloading jira issue link types... ✓
downloading jira priorities... ✓
downloading jira boards...: 100%|██████████| 3/3 [00:00<00:00,  3.71it/s]
downloading jira sprints: 100%|██████████| 4/4 [00:05<00:00,  1.50s/it]
downloading issue metadata... ✓
downloading jira issues: 100%|██████████| 312/312 [00:04<00:00, 74.86it/s]
downloading jira worklogs...  ✓
downloading jira custom field options... ✓
downloading jira statuses... ✓
Starting Git download for 1 provided git configurations
downloading github projects... ✓
downloading github users... ✓
downloading github repos...
downloading repos for Jellyfish-AI: 2repos [00:12,  6.20s/repos]
Done downloading repos!
downloading github commits on included branches...
downloading commits for branch develop in repo jellyfish (Jellyfish-AI): 463commits [00:04, 99.93commits/s]
downloading commits for branch main in repo jf_ingest (Jellyfish-AI): 30commits [00:00, 60.43commits/s]
Done downloading commits for included branches!
downloading github prs...
processing prs for jellyfish (81621092): 210prs [00:30,  6.82prs/s]
processing prs for jf_ingest (668351510): 4prs [00:02,  1.69prs/s]
Done downloading github PRs!
Compressing ./output/20231018_131022/diagnostics.json
Sending data to Jellyfish...
Starting 23 threads
Successfully uploaded jira_customfieldoptions.json.gz
Successfully uploaded jira_statuses.json.gz
Successfully uploaded jira_worklogs.json.gz
Successfully uploaded jira_linktypes.json.gz
Successfully uploaded jira_boards.json.gz
Successfully uploaded jira_issue_ids_downloaded.json.gz
Successfully uploaded status.json.gz
Successfully uploaded jira_resolutions.json.gz
Successfully uploaded diagnostics.json.gz
Successfully uploaded jira_projects_and_versions.json.gz
Successfully uploaded jira_sprints.json.gz
Successfully uploaded jira_issuetypes.json.gz
Successfully uploaded git_8v1crHqhmq/bb_users.json.gz
Successfully uploaded git_8v1crHqhmq/bb_projects.json.gz
Successfully uploaded jira_priorities.json.gz
Successfully uploaded jira_fields.json.gz
Successfully uploaded jira_board_sprint_links.json.gz
Successfully uploaded jira_users.json.gz
Successfully uploaded git_8v1crHqhmq/bb_commits.json.gz
Successfully uploaded git_8v1crHqhmq/bb_repos.json.gz
Successfully uploaded jira_issue_ids_deleted.json.gz
Successfully uploaded git_8v1crHqhmq/bb_prs.json.gz
Successfully uploaded jira_issues.json.gz
Successfully uploaded config.yml
Agent run succeeded: True
Successfully uploaded jf_agent.log
Successfully uploaded .done
Done!
```

Agent Upload from above run:
[diagnostics (17).json](https://github.com/Jellyfish-AI/jf_agent/files/13025610/diagnostics.17.json)
[jf_agent (45).log](https://github.com/Jellyfish-AI/jf_agent/files/13025611/jf_agent.45.log)

### CODE REVIEW

File: jf_agent/main.py [hunk 1]
```diff
@@ -14,10 +14,10 @@
 
 import requests
 import json
+import dotenv
```
Comment (jruel4) [inline jf_agent/main.py:17]: Q: Leftover from local dev?

**Comment by jruel4** [comment]
Q: Surely I just missed it - where is the diagnostics.json file being opened?

**Comment by gavinpitt-jf** [comment]
> Q: Surely I just missed it - where is the diagnostics.json file being opened?

The code for this didn't change, but the `diagnostics` module got moved, so it doesn't show up on this PR (I think)

The specific call is here: https://github.com/Jellyfish-AI/jf_agent/blob/680191622e95beca66bc332e22a8d4998c32d668/jf_agent/main.py#L161

**Comment by jruel4** [comment]
> > Q: Surely I just missed it - where is the diagnostics.json file being opened?
> 
> The code for this didn't change, but the `diagnostics` module got moved, so it doesn't show up on this PR (I think)
> 
> The specific call is here:
> 
> https://github.com/Jellyfish-AI/jf_agent/blob/680191622e95beca66bc332e22a8d4998c32d668/jf_agent/main.py#L161

Ahh yeah just found this - thank you!

File: jf_agent/main.py (hunk 1 above)
Comment (gavinpitt-jf) [reply jf_agent/main.py:17]: Whoops, looks like a double import. We have this import at line 3 on the master branch already.... I think I got my wire crossed a bit in this file

Good catch, I'll remove it

**Review by thingscouldbeworse — APPROVED** [review]
+1 assuming it's been run through a couple times and logs okay

**Review by jruel4 — APPROVED** [review]

**Pull request merged**